content/faqs/html-lazy-not-working.md

```diff
@@ -0,0 +1,112 @@
+>
+> Structural equality is used for `Int`, `Float`, `String`, `Char`, and `Bool`.
+> Reference equality is used for records, lists, custom types, dictionaries, etc.
+> Structural equality means that `4` is the same as `4` no matter how you produced those values. Reference equality means the actual pointer in memory has to be the same. Using reference equality is always cheap `O(1)`, even when the data structure has thousands or millions of entries. So this is mostly about making sure that using `lazy` will never slow your code down a bunch by accident. All the checks are super cheap!
+
+Sometimes it's not clear however when we're ending up with new references. Here are a few common pitfalls:
+
+
+### Gotcha: New values created in the view
+
+Here is a lazy that works:
+
+```
+import Html
+import Html.Lazy
+
+showUserLazy model =
+  Html.Lazy.lazy2 showUser model.user.firstname user.age
+
+showUser firstname age =
+  Html.div [] [
+    Html.text <| firstname ++ " is " ++ String.fromInt age ++ " years old."
+  ]
+```
+
+Here is a refactor that that doesn't:
+
+```
+import Html
+import Html.Lazy
+
+showUserLazy model =
+  let
+    userInfo = { firstname = model.user.firstname, age = model.user.age }
+  in
+  Html.Lazy.lazy showUser userInfo
+
+showUser userInfo =
+  Html.div [] [
+    Html.text <| userInfo.firstname ++ " is " ++ String.fromInt userInfo.age ++ " years old."
+  ]
+```
+
+Here the `userInfo` value is being created every time showUserLazy is called, so the .lazy always sees it as changed, because Records are compared by reference.
+
+
+### Gotcha: Using an anonymous view function instead of a named one
+
+What if we wanted to force a view to be computed once with some dynamic values, and never again?
+
+Maybe we could apply this trick, and use a static value that we ignore?
+
+```
+showUserLazy model =
+  Html.Lazy.lazy (\_ -> showUser model.user) "never changes, right?"
+```
+
+This doesn't work, because `lazy` also checks the function being passed in by reference!
+
+Here `(\_ -> showUser model.user)` is an anonymous function value that gets created every single time `showUserLazy` is called, so even though our argument is a static string, Html.Lazy is noticing our function is changing and busting the cache.
+
+
+
+### Hack: dynamic values and > 8 lazy arguments
+
+The largest arity function provided by `Html.Lazy` is `lazy8`.
+
+Usually this is more than sufficient, and if not, you may be able to combine multiple data points into a record in your model, and then pass the single record in to `lazy`.
+
+There may be situations where reworking your model for this is onerous. We can modify the code from our first gotcha to use serialisers and take advantage of Strings being compared by value:
+
+```
+import Html
+import Html.Lazy
+
+showUserLazy model =
+  let
+    userInfoString = userInfoToString { firstname = model.user.firstname, age = model.user.age }
```

Comment:
Probably worth having an example of `userInfoToString` and `userInfoFromString`.
Also, let's add a type annotation to make it clear what the type of `userInfoString` is.

Reply:
Yeah was worried the example would get too large, but maybe it's okay if in a section with the 3 groupings you've proposed below.